### ISSUE TITLE
Update logging in DNNClassifier to use tf.summary.scalar

### ISSUE THREAD

**Issue by feranick** [tensorflow/tensorflow#11041]
### System information
- I am using the stock DNNClassifier in contrib/learn. The warning is present in every instance where DNNClassifier is called.
- Present in OSX (Mac OS 10.12.5) and Linux (Ubuntu 16.04)
- **TensorFlow installed from (source or binary)**: Present both on binary and when compiled from source
- **TensorFlow version (use command below)**: v1.2.0 (release) and v1.2.0-1371-g97af82d53 1.2.0
- **Bazel version (if compiling from source)**: 0.5.1
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: Call the DNN classifier as indicated in https://www.tensorflow.org/api_docs/python/tf/contrib/learn/DNNClassifier

### Describe the problem
A warning for a deprecated feature is cluttering the logs when using the DNNClassifier. It is due to the use of the deprecated feature scalar_summary while logging. While this has been deprecated in 2016-11-30, it is still used in tensorflow/contrib/learn/python/learn/estimators/head.py:642
This bug report request for updating the the current tf.summary.scalar, as indicated. The usability of the product is much improved as a consequence.

### Logs
```
WARNING:tensorflow:From /opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/contrib/learn/python/learn/estimators/head.py:642: scalar_summary (from tensorflow.python.ops.logging_ops) is deprecated and will be removed after 2016-11-30.
Instructions for updating:
Please switch to tf.summary.scalar. Note that tf.summary.scalar uses the node name instead of the tag. This means that TensorFlow will automatically de-duplicate summary names based on the scope they are created in. Also, passing a tensor or list of tags to a scalar summary op is no longer supported.

```
### Source code
tensorflow/contrib/learn/python/learn/estimators/head.py:642
```
    # Uses the deprecated API to set the tag explicitly.
    # Without it, training and eval losses will show up in different graphs.
    logging_ops.scalar_summary(
        _summary_key(head_name, mkey.LOSS), weighted_average_loss)
```

**Comment by feranick** [comment]
From the comments to the line in code in question, the choice of retaining the deprecated feature seems to be intentional:
tensorflow/contrib/learn/python/learn/estimators/head.py:642
    ```
# Uses the deprecated API to set the tag explicitly.
    # Without it, training and eval losses will show up in different graphs.
    logging_ops.scalar_summary(
        _summary_key(head_name, mkey.LOSS), weighted_average_loss)
```
While I understand the rationale, the resulting cluttering of the log makes it rather difficult to read.

**Comment by skye** [comment]
@ispirmustafa can you comment on this or redirect to someone else? Thanks!

**Comment by ispirmustafa** [comment]
Yes we keep it the deprecated version. This is fixed in the core version of DNNClassifier. (tf.estimator.DNNclassifier) which will be available with 1.3 or at head.
